appwrite/appwrite
# Implement CPU and Memory runtime controls

## What does this PR do?

This PR implements CPU and Memory controls for Runtimes, adding new `cpus` and `memory` attributes onto the functions document and it's relevant API endpoints. `_APP_FUNCTIONS_CPUS` and `_APP_FUNCTIONS_MEMORY` are now utilised as the max amount of memory and CPUs that can be allocated to a function.

### To be implemented
- [x] Plan Differentiators in API Parameters
- [x] Using `_APP_FUNCTIONS_CPUS` and `_APP_FUNCTIONS_MEMORY` as limiters and to reflect that in the API
- [x] Implementing the API in console
- [x] Fixing the issue we're having with TS Enums having numeric names

## Test Plan
1. Create a function and execution then check it's stats
2. Change the runtime memory, rebuild and check it's stats again
3. Change the runtime CPU, rebuild and check it's stats one more time

## Related PRs and Issues

- https://github.com/appwrite/console/pull/1138

## Checklist
